doc/programming/parsing.rst

```diff
@@ -469,6 +469,18 @@ There are three locations where hooks can be implemented:
   type accordingly. They provide a powerful mechanism to extend a
   predefined unit without changing any of its code.
 
+If multiple implementations are provided for the same hook, by default
+it remains undefined in which order they will execute. If a particular
+order is desired, you can specify priorities for your hook
+implementations::
+
+      on Foo::v priority=-5 { ... }
+      on Foo::v priority=5 { ... }
+
+Implementations then execute in order of their priorities: The lower a
+priority value, the earlier it will execute. If not specified, a
```

Comment:
Ah I see, maybe disregard my comment from above on possibly leaving them unset by default is not the behavior we want. Do you have a specific use case for hooks with negative priority in mind? -- maybe something worth adding an example for.